pkg/plugin/api/server.go

```diff
@@ -379,14 +380,18 @@ func (c *grpcServer) ForceFrontendUpdate(ctx context.Context, _ *proto.Empty) (*
 	return &proto.Empty{}, nil
 }
 
-// SendAlert sends an alert
+// SendAlert sends an alert. This method is deprecated and will be removed in a future release
```

Comment:
Just to be clear, we'd just need to deprecate it from the server side, right? As in, we could modify `SendAlert()` in `client.go` to call `client.SendEvent()`, so that plugins sending alerts won't need to be rewritten. I feel like `SendAlert()` is a good utility to have client-side, since it's widely usable. Were it removed, any plugins would need to re-implement the same code, and it would look almost identical to what we have already anyways. 

https://github.com/vmware-tanzu/octant/blob/f2b531ca649e23e711f9118019ea185e8412626f/pkg/plugin/api/client.go#L242-L255

Reply:
The idea is to remove it entirely because it is redundant (and we can do so relatively safely as it is pre 1.0). Otherwise one could argue one should have similar methods for each event type.

The tradeoff is abstracting away knowledge of the required `action.Payload` keys which is something that could be addressed by creating payload helpers for various events.

Reply:
Makes sense, some sort of action/event helpers would be super useful for plugins